docs/content/post/developer-documentation.md

```diff
@@ -0,0 +1,256 @@
+            "name": "eth0",
+            "mac": "1a:9b:80:a1:1e:99",
+            "sandbox": "/var/run/netns/testing"
+        }
+    ],
+    "ips": [
+        {
+            "version": "4",
+            "interface": 0,
+            "address": "192.168.10.99/24",
+            "gateway": "192.168.10.254"
+        }
+    ],
+    "dns": {}
+```
+
+Now we can just enter that namespace and check that the macvlan interface and
+the iptables rules were applied.
+
+```bash
+$ sudo ip netns exec testing ip a
+1: lo: <LOOPBACK> mtu 65536 qdisc noop state DOWN group default qlen 1000
+    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
+2: eth0@if6: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default
+    link/ether 1a:9b:80:a1:1e:99 brd ff:ff:ff:ff:ff:ff link-netnsid 0
+    inet 192.168.10.99/24 brd 192.168.10.255 scope global eth0
+       valid_lft forever preferred_lft forever
+    inet6 fe80::189b:80ff:fea1:1e99/64 scope link
+       valid_lft forever preferred_lft forever
+```
+
+```bash
+$ sudo ip netns exec testing iptables -t nat -L
+Chain PREROUTING (policy ACCEPT)
+target     prot opt source               destination
+DNAT       all  --  anywhere             anywhere             to:10.0.3.0
+
+Chain INPUT (policy ACCEPT)
+target     prot opt source               destination
+
+Chain OUTPUT (policy ACCEPT)
+target     prot opt source               destination
+
+Chain POSTROUTING (policy ACCEPT)
+target     prot opt source               destination
+SNAT       all  --  anywhere             anywhere             to:192.168.10.99
+
+```
+
+## Using quay.io/openshift image
+
+In order to deploy this into a cluster, you'll have to either compile the
+inclusded `Dockerfile` or rely into the preexising images that are shipped into
+openshift's quay
+(https://quay.io/repository/openshift/origin-egress-router-cni).
+
+If deploying an openshift cluster, we're already deploying this image alongside
+CNO's multus' daemonsets, so you've got nothing to do. Only if you want to
+change the default image to yours, you'll have to override the
+`EGRESS_ROUTER_CNI_IMAGE` value with the reference to your image.
+
+If you don't want to use CNO at all, you can still install the egress router
+cni by manually copying it to the CNI binary path, which should be
+`/var/lib/cni/bin/`
+
+### How to use the egress-router-cni
+
+Egress Router CNI aims to match openshift-sdn functionality, please refer to
+[this nice openshift blog article](https://www.openshift.com/blog/accessing-external-services-using-egress-router)
+for further details on it. In order to make things easier for the legacy router
+users, so far the `redirect` mode is supported in egress-router-cni.
+
+In openshift-sdn, the egress router was implemented by adding an annotation to
+allow a pod to request a macvlan interface. In order to avoid repeating this
+behavior in ovn-kubernetes, we'd be requesting such interface using multus to
+ensure feature-parity with openshift-sdn.
```

Comment:
"In openshift-sdn, the egress router was implemented by adding an annotation to
allow a pod to request a macvlan interface. In contrast, the new egress-router
implementation uses CNI to create the macvlan interface and so can be used
with any network plugin."

(ie, it's not an ovn-kubernetes feature. It's a network-plugin-agnostic feature.)